### ISSUE TITLE
Error in SearchVenues

### ISSUE THREAD

**Issue by michaelpimentel** [TICLAB/SharpSquare#7]
Today my application started showing error 410 message in the search operation. I did a little debugging and find it would be something new requirement of the foursquare api related to parameter "V" . So resolve to upgrade SharpSquare to the latest version that is here in github, however I am having the error below:

Msg:
Cannot deserialize the current JSON array (e.g. [1,2,3]) into type 'FourSquare.SharpSquare.Entities.FourSquareEntityItems`1[FourSquare.SharpSquare.Entities.Venue]' because the type requires a JSON object (e.g. {"name":"value"}) to deserialize correctly.

To fix this error either change the JSON to a JSON object (e.g. {"name":"value"}) or change the deserialized type to an array or a type that implements a collection interface (e.g. ICollection, IList) like List<T> that can be deserialized from a JSON array. JsonArrayAttribute can also be added to the type to force it to deserialize from a JSON array.

Path 'response.neighborhoods', line 1, position 50.

Here:
 public List<Venue> SearchVenues(Dictionary<string, string> parameters)
        {
            FourSquareEntityItems<Venue> venues = GetSingle<FourSquareEntityItems<Venue>>("/venues/search", parameters, true).response["groups"];

```
        return venues.items;
    }
```

**Comment by petrmanek** [comment]
Same error here. Any progress on the issue yet?

My code:
```
var fsq = new SharpSquare(fsqAppId, fsqAppSecret);
var search = fsq.SearchVenues(new Dictionary<string, string>
{
    { "ll", "50.088314, 14.416286" },
    { "intent", "checkin" }
});
```

Exception with Stack Trace:
```
Cannot deserialize the current JSON array (e.g. [1,2,3]) into type 'FourSquare.SharpSquare.Entities.FourSquareEntityItems`1[FourSquare.SharpSquare.Entities.Venue]' because the type requires a JSON object (e.g. {"name":"value"}) to deserialize correctly.
To fix this error either change the JSON to a JSON object (e.g. {"name":"value"}) or change the deserialized type to an array or a type that implements a collection interface (e.g. ICollection, IList) like List<T> that can be deserialized from a JSON array. JsonArrayAttribute can also be added to the type to force it to deserialize from a JSON array.
Path 'response.venues', line 1, position 43.

   v Newtonsoft.Json.Serialization.JsonSerializerInternalReader.EnsureArrayContract(JsonReader reader, Type objectType, JsonContract contract)
   v Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateList(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, Object existingValue, String id)
   v Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   v Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateDictionary(IDictionary dictionary, JsonReader reader, JsonDictionaryContract contract, JsonProperty containerProperty, String id)
   v Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateObject(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   v Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   v Newtonsoft.Json.Serialization.JsonSerializerInternalReader.SetPropertyValue(JsonProperty property, JsonConverter propertyConverter, JsonContainerContract containerContract, JsonProperty containerProperty, JsonReader reader, Object target)
   v Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateObject(Object newObject, JsonReader reader, JsonObjectContract contract, JsonProperty member, String id)
   v Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateObject(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   v Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   v Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Deserialize(JsonReader reader, Type objectType, Boolean checkAdditionalContent)
   v Newtonsoft.Json.JsonSerializer.DeserializeInternal(JsonReader reader, Type objectType)
   v Newtonsoft.Json.JsonSerializer.Deserialize(JsonReader reader, Type objectType)
   v Newtonsoft.Json.JsonConvert.DeserializeObject(String value, Type type, JsonSerializerSettings settings)
   v Newtonsoft.Json.JsonConvert.DeserializeObject[T](String value, JsonSerializerSettings settings)
   v Newtonsoft.Json.JsonConvert.DeserializeObject[T](String value)
   v FourSquare.SharpSquare.Core.SharpSquare.GetSingle[T](String endpoint, Dictionary`2 parameters, Boolean unauthenticated)
   v FourSquare.SharpSquare.Core.SharpSquare.SearchVenues(Dictionary`2 parameters)
```

**Comment by petrmanek** [comment]
Okay. I've got few things to say.

Firstly, my error was caused by old NuGet package. Please update it to the latest version of this repo **ASAP**. As a workaround, I've had to clone your repo as a submodule and manually reference it in my own solution.

Unfortunately, that still didn't fix the issue for me. The error just moved from `GetSingle<T>` to `GetMultiple<T>`. However, I've been lucky to come across [a bug report](https://github.com/TICLAB/SharpSquare/issues/10) which provided an ugly, yet effective solution.

To sum up, I've had to patch `SharpSquare.cs` with this code:
```
// In GetMultiple<T>(string, Dictionary<string, string>, bool):
// Just under 'string json = Request(...', add these lines:
int index = json.IndexOf(",\"confident");
if (index > 0)
    json = json.Substring(0, index);
json = json + "}}";
```

Since this patch is really temporary and may stop working at any time, I encourage somebody to take on this issue and fix it permanently.

Cheers!
PM
